### PR TITLE
use param_ref in nla_solver

### PR DESCRIPTION
**Pull request by levnach**

### CODE REVIEW

File: src/math/lp/nla_tangent_lemmas.cpp [hunk 1]
```diff
@@ -186,7 +186,7 @@ tangents::tangents(core * c) : common(c) {}
 void tangents::tangent_lemma() {
     factorization bf(nullptr);
     const monic* m = nullptr;
-    if (c().m_nla_settings.run_tangents && c().find_bfc_to_refine(m, bf)) {
```
Comment (NikolajBjorner) [inline src/math/lp/nla_tangent_lemmas.cpp:189]: you still have to remove fields from m_nla_settings

File: package-lock.json [hunk 2]
```diff
@@ -1,27 +1,6 @@
 {
+  "name": "z3",
```
Comment (NikolajBjorner) [inline package-lock.json:2]: this file should not be in the pull request

**Comment by NikolajBjorner** [comment]
the code that uses grobner_quota from nla_settings.h should be updated to not update the parameter itself. See arith branch.
The arith branch leaves a single field left in nla_settings. It can maybe be eliminated too.

File: src/sat/smt/arith_internalize.cpp [hunk 3]
```diff
@@ -67,23 +67,6 @@ namespace arith {
                 m_nla->push();
             }
             smt_params_helper prms(s().params());
```
Comment (NikolajBjorner) [inline src/sat/smt/arith_internalize.cpp:69]: this line also goes away

File: src/smt/theory_lra.cpp [hunk 4]
```diff
@@ -274,23 +274,6 @@ class theory_lra::imp {
             };
             m_nla->set_relevant(is_relevant);
             smt_params_helper prms(ctx().get_params());
```
Comment (NikolajBjorner) [inline src/smt/theory_lra.cpp:276]: this line also goes away

File: package-lock.json (hunk 2 above)
Comment (levnach) [reply package-lock.json:2]: It seems @NikolajBjorner has added it 
C:\dev\z3>git log .\package-lock.json
commit b3f5cbe3f4f1727b8df1a1e1cbc9438f2f85466f
Author: Nikolaj Bjorner <nbjorner@microsoft.com>
Date:   Thu Aug 17 02:09:27 2023 +0000

    add parameters

    Signed-off-by: Nikolaj Bjorner <nbjorner@microsoft.com>

Comment (levnach) [reply package-lock.json:2]: Should I restore it to the previous version?

**Comment by NikolajBjorner** [comment]
I deleted it. It seems introduced [on Wed Jul 06 2022](https://github.com/Z3Prover/z3/commit/9d9414c1110bc65108202cdcc473c703d12b54f1)
for some random reason.

**Pull request merged**